plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -144,31 +147,52 @@ instance Show fun => Pretty (ExBudgetCategory fun) where
 instance ExBudgetBuiltin fun (ExBudgetCategory fun) where
     exBudgetBuiltin = BBuiltinApp
 
-{- Note [Arities in VBuiltin]
-The VBuiltin value below contains two copies of the arity (list of
-TypeArg/TermArg pairs) for the relevant builtin.  The second of these
-is consumed as the builtin is instantiated and applied to arguments,
-to check that type and term arguments are interleaved correctly.  The
-first copy of the arity is left unaltered and only used by
-dischargeCekValue if we have to convert the frame back into a term
-(see mkBuiltinApplication).  An alternative would be to look up the
-full arity in mkBuiltinApplication, but that would require a lot of
-things to be monadic (including the PrettyBy instance for CekValue,
-which is a problem.)
+{- Note [Instances for BuiltinRuntime]
+We need to be able to print 'CekValue's and for that we need a 'Show' instance for 'BuiltinRuntime',
+but functions are not printable and hence we provide a dummy instance. Same applies to 'Eq'.
+
+We could define
+
+    instance Eq (BuiltinRuntime term) where
+        _ == _ = True
+
+however this is very unsafe (as two different functions can be, well, different). The reason
+why
+
+    instance Eq (BuiltinRuntime (CekValue uni fun))
+        _ == _ = True
+
+is not as unsafe is because the 'VBuiltin' constructor stores the term representation of the
+partially applied builtin and so we get actual equality from there, which is why it's fine to
+ignore the partially applied builtin itself.
```

Comment:
I'm just being lazy. I can provide a direct instance instead of deriving it. Doesn't apply to `Show` though, which is much easier to derive than to define manually. Should I keep `Show` and define `Eq`?

Reply:
Turns out we don't need that `Eq` instance at all, so I dropped it.